kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/WatchEventsListener.java

```diff
@@ -0,0 +1,119 @@
+import org.slf4j.Logger;
+
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+class WatchEventsListener extends WebSocketListener {
+  private final AtomicReference<WebSocket> webSocketRef = new AtomicReference<>();
+  private final ScheduledExecutorService executor = Executors.newSingleThreadScheduledExecutor();
+  private final AttributeSet attributeSet;
+  private final Context context;
+  private final Set<WatchEventsListener> watchEventListenerList;
+  private final Logger logger;
+
+  public WatchEventsListener(Context context, AttributeSet attributeSet, final Set<WatchEventsListener> watchEventListenerList, Logger logger) {
+    this.logger = logger;
+    this.context = context;
+    this.attributeSet = attributeSet;
+    this.watchEventListenerList = watchEventListenerList;
+  }
+
+  public boolean attributeMatches(AttributeSet set) {
+    return set.matches(attributeSet);
+  }
+  @Override
+  public void onOpen(WebSocket webSocket, Response response) {
+    webSocketRef.set(webSocket);
+  }
+
+  @Override
+  public void onClosing(WebSocket webSocket, int code, String reason) {
+    webSocketRef.get().close(code, reason);
+  }
+
+  @Override
+  public void onClosed(WebSocket webSocket, int code, String reason) {
+    if (webSocketRef.get() != null){
+      webSocketRef.get().close(code, reason);
+    }
+    executor.shutdown();
+    try {
+      executor.awaitTermination(1, TimeUnit.MINUTES);
+    } catch (InterruptedException e) {
+      logger.debug(e.getLocalizedMessage());
+      Thread.currentThread().interrupt();
+    }
+    watchEventListenerList.remove(this);
+  }
+
+  @Override
+  public void onFailure(WebSocket webSocket, Throwable t, Response response) {
+    webSocket.close(1000, t.getMessage());
+    executor.shutdown();
+    try {
+      executor.awaitTermination(1, TimeUnit.MINUTES);
+    } catch (InterruptedException e) {
+      logger.debug(e.getLocalizedMessage());
+      Thread.currentThread().interrupt();
+    }
+    watchEventListenerList.remove(this);
+  }
+
+
+  public void sendWebSocketResponse(String object, String eventType) {
+    WebSocketMessage message = toWebSocketMessage(context, new WatchEvent(Serialization.unmarshal(object, KubernetesResource.class), eventType));
+    executor.schedule(() -> webSocketRef.get().send(message.getBody()), message.getDelay(), TimeUnit.SECONDS);
+  }
+
+  private static WebSocketMessage toWebSocketMessage(Context context, Object content) {
+    return toWebSocketMessage(context, content, true);
+  }
+
+  private static WebSocketMessage toWebSocketMessage(Context context, Object content, Boolean toBeRemoved) {
+    return toWebSocketMessage(context,0L, content, toBeRemoved);
+  }
+
+  private static WebSocketMessage toWebSocketMessage(Context context, Long delay, Object content, Boolean toBeRemoved) {
```

Comment:
Any specific reason you have made these methods `static`?

Reply:
What's the reason not make them static if they are not affecting the state of the object and they don't require any of the declared fields (helper-methods)?

Reply:
Umm, ok